disnake/interactions/base.py

```diff
@@ -624,7 +624,13 @@ async def defer(self, *, ephemeral: bool = False) -> None:
         -----------
         ephemeral: :class:`bool`
             Indicates whether the deferred message will eventually be ephemeral.
-            This only applies for interactions of type :attr:`InteractionType.application_command`.
+            This only applies for interactions of type :attr:`InteractionType.application_command` or when ``with_message`` is True
```

Comment:
Would it be better to refer to the specific interaction classes (`ApplicationCommandInteraction`, `MessageInteraction`) here, instead of `InteractionType.*`?

Reply:
it currently refers to InteractionType on master, so I don't want to change it since that may be more confusing. 

I'm not sure, I'll leave it up to you.

Reply:
I feel like referring to the interaction classes would be less confusing, since you generally don't use `InteractionType` much unless you implement your own handlers.
On the other hand, `InteractionType` is more accurate, so let's leave it like this for now; can always revisit it later.